### PR TITLE
ensure highlight dotnet 4 sdk installs dependencies correctly

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Rebuilds the Highlight .NET4 SDK to be Windows Visual Studio 2022 .NET Framework compatible.
API interface to using remains the same; the NuGet package has been fixed.

## How did you test this change?

.NET Framework 4.8 Visual Studio 2022 MVC app
image

.NET Core 8 Visual Studio 2022 MVC app
image

## Are there any deployment considerations?

NuGet package deployed

## Does this work require review from our design team?

no

### CODE REVIEW

File: sdk/highlight-dotnet4/OpenTelemetry.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public static Dictionary<string, object> GetResourceAttributes()
                 ["highlight.project_id"] = Cfg.ProjectId,
                 ["service.name"] = Cfg.ServiceName,
                 ["telemetry.distro.name"] = "Highlight.ASP4",
-                ["telemetry.distro.version"] = "0.1.7",
+                ["telemetry.distro.version"] = "0.2.0-beta.7",
```
Comment (ccschmitz) [inline sdk/highlight-dotnet4/OpenTelemetry.cs:110]: Think the warnings about `Serilog` are relevant?

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -33,14 +33,14 @@ jobs:
               run: dotnet pack -c Release -o out
 
             - name: Push generated package to NuGet registry
-              if: github.ref == 'refs/heads/main'
-              run: dotnet nuget push out/*.nupkg -k $NUGET_AUTH_TOKEN -s https://api.nuget.org/v3/index.json
+              # if: github.ref == 'refs/heads/main'
```
Comment (ccschmitz) [inline .github/workflows/dotnet.yml:36]: Just curious, is this just a temporary workaround to force republishing?

**Review by ccschmitz — APPROVED** [review]

File: .github/workflows/dotnet.yml (hunk 2 above)
Comment (Vadman97) [reply .github/workflows/dotnet.yml:36]: yup - have a TMP commit d1f9cf16a7e887cff6321b1d40ea1cacbffcee09 that i'll revert before merging

File: sdk/highlight-dotnet4/OpenTelemetry.cs (hunk 1 above)
Comment (Vadman97) [reply sdk/highlight-dotnet4/OpenTelemetry.cs:110]: which ones?

**Pull request merged**